src/dialects/mssql/query-generator.js

```diff
@@ -976,12 +976,39 @@ class MSSQLQueryGenerator extends AbstractQueryGenerator {
       // TODO: document why this is adding the primary key of the model in ORDER BY
       //  if options.include is set
       if (!options.order || options.order.length === 0 || options.include && orders.subQueryOrder.length === 0) {
-        const tablePkFragment = `${this.quoteTable(options.tableAs || model.name)}.${this.quoteIdentifier(model.primaryKeyField)}`;
+        let primaryKey = model.primaryKeyField;
+
+        const tablePkFragment = `${this.quoteTable(options.tableAs || model.name)}.${this.quoteIdentifier(primaryKey)}`;
+        const aliasedAttribute = (options.attributes || []).find(attr => Array.isArray(attr)
```

Comment:
This is a copy from the generic query generator. I was about to refactor this function but then wondered if `_getAliasForField` should somehow take care of this logic. I'm still not 100% sure what we are even doing here but I supposed, that this piece of code combined with getAliasForField is basically catering for 1) model definition based aliases (using the field property?) and 2) aliases that are introduced during the query calls (e.g. findAll).

Maybe instead of doing this manually, we should have a bit of code inside of _getAliasForField that is checking the options and extracts the call alias from them?